### ISSUE TITLE
Removal of relationship from attributes fails when property naming strategy is used

### ISSUE THREAD

**Issue by shyamalprasad** [jasminb/jsonapi-converter#130]
If an `ObjectMapper` with a `KEBAB_CASE` property naming strategy is used we find that any relationship members that have camel cased names do not get removed from the `attributes` of the JSON generated. We will end up with two copies of the member: one (wrongly) in the attributes, and the other (as expected) in the relationship and/or included sections.

For example, using a declaration like this:

```
@Relationship("integer-id-relationship")
private IntegerIdResource integerIdResource;
```

will result in a extra `integer-id-relationship` value in the `attributes`, and also the expected relationship as expected, when used with a `KEBAB_CASE` property naming strategy.

The root cause is that the code that removes the attribute does not use the configured property naming strategy for the `objectMapper` of the converter. In fact, this bug will manifest itself for any relationship member whose name is modified by the property naming strategy in use.

A test case that demonstrates this problem is available: https://github.com/jasminb/jsonapi-converter/pull/128/commits/83a562020a58d58563acd533d73cab56e9dccd24

A fix is available in https://github.com/jasminb/jsonapi-converter/pull/128

**Comment by GuyPaddock** [comment]
The fix in #128 doesn't work for me in the latest `0.9-SNAPSHOT` with a `PropertyNamingStrategy` of `SnakeCaseStrategy`. I'm using the Jackson annotation method of setting it, though, which I'm guessing makes the difference.

See the [attached source file](https://github.com/jasminb/jsonapi-converter/files/1475824/User.java.zip) for an example. The example produces this output:
```JSON
{
  "data" : {
    "type" : "user",
    "id" : "8675309",
    "attributes" : {
      "name" : "Bob",
      "home_location" : {
        "id" : 5,
        "name" : "Switzerland"
      }
    },
    "relationships" : {
      "home_location" : {
        "data" : {
          "type" : "location",
          "id" : "5"
        }
      }
    }
  }
}
```

Interestingly, if I add a `@JsonIgnore` annotation to the field, that prevents the relationship from appearing in `attributes`, as follows:

```Java
public class User {
  // ...
  @Relationship("home_location")
  @JsonIgnore
  private Location homeLocation;
  // ...
}
```
Produces:
```JSON
{
  "data" : {
    "type" : "user",
    "id" : "8675309",
    "attributes" : {
      "name" : "Bob"
    },
    "relationships" : {
      "home_location" : {
        "data" : {
          "type" : "location",
          "id" : "5"
        }
      }
    }
  }
}
```

Not great, but it's workable until there's a better fix in place.

**Comment by shyamalprasad** [comment]
@GuyPaddock sorry, I just got to this. The problem you identify is real, but I'd suggest opening a new issue to track it (it actually pre-dates #130 in some way).

Here is what is going on: the conversion is always executed using the object mapper in the ResourceConverter. The naming strategy used to remove relationship subtrees uses the naming strategy of this object mapper and **not** the naming strategy for the object being serialized as specified by a Jackson annotation!.

For your case, a simple work around would be to set the naming strategy on the Object mapper, and pass it to the ResourceConverter constructor explicitly. That _should_ work unless you are using different naming strategies for different objects (which a real fix should handle).

Let me know how it goes (I'm not the maintainer or an active contributor to this library, just an interested user).

**Comment by marceloverdijk** [comment]
I think this one is fixed as part of #165 / #166.

**Comment by jasminb** [comment]
Closing as issue was fixed.
